app/3.0/docs/es6.md

```diff
@@ -1,46 +1,145 @@
+import './my-app-element.js';
+```
+Modules can also be loaded from HTML using `<script type="module">`
+
+```js
+<script type="module" src="./main-module.js">
+</script>
+
+<!-- modules can be inline, too -->
+<script type="module">
+import {fooBar} from './my-module.js';
+fooBar('baz');
+</script>
+```
+
+There are a number of online primers on JavaScript modules if you're not familiar with them. 
+
+*   [ECMAScript modules in browsers](https://jakearchibald.com/2017/es-modules-in-browsers/) by Jake Archibald provides a
+    good overview of modules.
+*   [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) and 
+    [export](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export) pages on MDN 
+    provide a quick reference on module syntax.
+*   [Modules chapter](http://exploringjs.com/es6/ch_modules.html) from Axel Rauschmayer's Exploring ES6.
+
+As an addition to the standard ES6 module syntax, Polymer supports importing modules by name, as described
+in the next section.
+
+### Module specifiers
+
+The browser only accepts on kind of module specifier in an `import` statement: a URL, which must
+be either fully-qualified, or a path starting with `/`, `./` or `../`. This works fine for importing 
+application-specific elements and modules:
+
+```js
+import './my-app-element.js';
+```
+
+However, it's challenging when you're writing a reusable component, and you want to import a peer dependency 
+installed using npm. The path may vary depending on how the components are installed. So Polymer supports 
+the use of Node-style named import specifiers:
+
+```js
+import {PolymerElement} from '@polymer/polymer/polymer-element.js';
+```
+
+Where `@polymer/polymer` is the _name_ of the npm package. (This style of specifier is sometimes called a 
+"bare module specifier".)
+
+These module specifiers need to be transformed to paths before they're served to the browser. The Polymer CLI can transform 
+them at build time, and the Polymer development server can transform them at runtime, so you can test code without a build step.
+Many third-party build tools, like WebPack and Rollup also support named modules.
+
+### Dynamic imports
+
+Dynamic imports is a language feature that enables lazy-loading of resources, replacing the functionality 
+that was previously used for this in Polymer 2.x (`Polymer.importHref`). 
+
+The import operator acts like a function, and returns a `Promise`:
+
+```js
+import('my-view1.js').then((MyView1Module) => {
+  console.log("MyView1 loaded");
+}).catch((reason) => {
+  console.log("MyView1 failed to load", reason);
+});
+```
+
+The latest versions of Chrome and Safari support dynamic imports. For other browsers, the Polymer build tools can transform
+dynamic imports into AMD modules. The Polymer development server can do this transformation on the fly.
+
+More on dyanmic import, see:
```

Comment:
More -> For more

"dyanmic" -> "dynamic"

"import" -> "imports" or "`import()`"